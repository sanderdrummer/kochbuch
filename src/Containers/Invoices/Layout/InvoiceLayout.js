import React, {Component} from 'react';
import {Route} from 'react-router';
import InvoiceOverview from '../Overview/InvoiceOverview';
import InvoiceAdderContainer from '../Adder/InvoiceAdder';
import Aux from '../../../Components/Ui/Auxiliary/auxiliary';

class InvoiceLayout extends Component {

    render() {
        return (
            <Aux>
                <Route path={this.props.match.url} exact component={InvoiceOverview} />
                <Route path={this.props.match.url + '/add'} exact component={InvoiceAdderContainer} />
            </Aux>
        )
    }
}


export default (InvoiceLayout);