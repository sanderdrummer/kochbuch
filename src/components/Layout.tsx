import { CommonProps } from './CommonProps'

export const HeightWrapper = (props: CommonProps) => {
  return <div class={`min-h-screen ${props?.class ?? ''}`}>{props.children}</div>
}
