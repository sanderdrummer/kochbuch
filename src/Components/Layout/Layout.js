import React from 'react';
import {NavLink, Route} from 'react-router-dom';
import ShoppingListContainer from '../../Containers/ShoppingList/ShoppingListLayout';
import InvoiceLayout from '../../Containers/Invoices/Layout/InvoiceLayout';
import UserAuthForm from '../../Containers/User/UserAuthForm/UserAuthForm';

const Layout = (props) => (
    <div>
        <nav className="navbar has-shadow" role="navigation" aria-label="main navigation">
            <div className="container">
                <div className="navbar-tabs">
                    <NavLink activeClassName="is-active"
                             className="navbar-item is-tab"
                             exact
                             to="/lists">Listen</NavLink>
                    <NavLink activeClassName="is-active"
                             className="navbar-item is-tab"
                             exact
                             to="/recipies">Rezepte</NavLink>
                    <NavLink activeClassName="is-active"
                             className="navbar-item is-tab"
                             exact
                             to="/invoices">Rechnungen</NavLink>
                </div>
            </div>
        </nav>
        <div className="section">
            <div className="container">
                <Route path="/lists" component={ShoppingListContainer}/>
                <Route path="/invoices" component={InvoiceLayout}/>
            </div>
        </div>
        <div className="section">
            <UserAuthForm/>
        </div>
    </div>
);


Layout.propTypes = {};

export default Layout;