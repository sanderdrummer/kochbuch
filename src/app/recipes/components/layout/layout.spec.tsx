import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Layout from './layout'; 

it('should render layout', () => {
    const wrapper = shallow(<Layout />);
    expect(toJson(wrapper)).toMatchSnapshot();
});