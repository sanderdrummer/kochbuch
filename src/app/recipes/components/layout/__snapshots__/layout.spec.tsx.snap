// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render layout 1`] = `
<React.Fragment>
  <Connect(Selection) />
  &gt;
  <Connect(Selection) />
</React.Fragment>
`;
